### PR TITLE
Move from flask_restful to flask_restx. Replace error dictionaries with abort

### PR DESCRIPTION
**Pull request by maxachis**
#### Fixes

* None. Quality of life enhancement.
* Obviates https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/276

#### Description

* Replaces all instances of returning error dictionaries with [flask_restx.abort](https://flask-restx.readthedocs.io/en/latest/errors.html) in multiple resources files. This refactoring enhances code readability and simplifies error handling. It also ensures that HTTP errors are handled in a more standardized way across the application.
* Resource class and `Api` usage in `app.py` has transitioned from `flask_restful` to `flask_restx`, which supports automated swagger documentation and has more active support. 

#### Testing

* None. Testing will be encompassed in expanded tests for all modules.

#### Performance

* No expected performance impact. 

#### Docs

* Not applicable.

### CODE REVIEW

File: middleware/security.py [hunk 1]
```diff
@@ -80,25 +82,27 @@ def decorator(*args, **kwargs):
             if len(authorization_header) >= 2 and authorization_header[0] == "Bearer":
                 api_key = request.headers["Authorization"].split(" ")[1]
                 if api_key == "undefined":
-                    return {"message": "Please provide an API key"}, 400
+                    abort(code=400, message="Please provide an API key")
```
Comment (joshuagraber) [inline middleware/security.py:85]: @maxachis Is this going to change the shape of the error objects we receive on the client?

**Review by joshuagraber — APPROVED** [review]
Looks good, just one question on how this changes the error response objects.

File: middleware/security.py (hunk 1 above)
Comment (maxachis) [reply middleware/security.py:85]: @joshuagraber Based on my tests, yes, but only somewhat, and possibly not in a way that interferes. The `response.json` will now include a "http_status_code" key-value pair in addition to the "message" key-value pair. 

So `{'message': 'An unexpected error occurred'}` will become `{'http_status_code': 500, 'message': 'An unexpected error occurred'}`

Comment (joshuagraber) [reply middleware/security.py]: No problem @maxachis thank you.

**Pull request merged**